internal/url/url.go

```diff
@@ -0,0 +1,116 @@
+package url
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+)
+
+func Sanitize(uri string) (string, error) {
+	// URL sanitanization for host argument
+	parsedURI, err := url.ParseRequestURI(uri)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse URI: %w", err)
+	}
+	// Remove any trailing slashes
+	parsedURI.Path = strings.TrimSuffix(uri, "/")
+	// Collapse any doubled slashes
+	parsedURI.Path = strings.ReplaceAll(uri, "//", "/")
```

Comment:
Ah yeah, that was a mistake. Let me fix that.

Reply:
Should be fixed now.

Reply:
Looks like it works.

Reply:
On a side note related to the SMD thing, it looks like the node information gets added despite the 500 error because if I run the same command immediately after, I get a 409:
```json
{"level":"error","error":"returned status code 409 when adding endpoint","time":"2024-08-14T19:54:12Z","message":"failed to add Redfish endpoint"}
```

Reply:
Going to resolve this thread since the change was made and tested.